### PR TITLE
C#: Change IsARM to Apple silicon check

### PR DESCRIPTION
**Pull request by tamasvajk**
The autobuilder was checking if the process was on ARM macos, and executed `dotnet msbuild` instead of `msbuild` if so. However, we execute an x64 dotnet binary on an ARM machine, so the execution platform reported is x64, which in turn leads to running `msbuild`, and not `dotnet msbuild`. Instead of checking the execution platform, we're now checking the host operating system, as that's relevant to whether the mono installation has msbuild or not.

### CODE REVIEW

File: csharp/autobuilder/Semmle.Autobuild.Shared/MsBuildRule.cs [hunk 1]
```diff
@@ -15,12 +15,12 @@ internal static class MsBuildCommandExtensions
         /// <returns></returns>
         public static CommandBuilder MsBuildCommand(this CommandBuilder cmdBuilder, IAutobuilder<AutobuildOptionsShared> builder)
         {
-            var isArmMac = builder.Actions.IsMacOs() && builder.Actions.IsArm();
+            var IsRunningOnAppleSiliconMac = builder.Actions.IsMacOs() && builder.Actions.IsRunningOnAppleSilicon();
```
Comment (criemen) [inline csharp/autobuilder/Semmle.Autobuild.Shared/MsBuildRule.cs:18]:
```suggestion
            var IsRunningOnAppleSiliconMac = builder.Actions.IsRunningOnAppleSilicon();
```

Comment (criemen) [reply csharp/autobuilder/Semmle.Autobuild.Shared/MsBuildRule.cs:18]: maybe: even inline this into the ?: operator

**Review by criemen — APPROVED** [review]
One nit, otherwise LGTM - from my perspective, you can merge this as-is.

File: csharp/autobuilder/Semmle.Autobuild.Shared/MsBuildRule.cs (hunk 1 above)
Comment (tamasvajk) [reply csharp/autobuilder/Semmle.Autobuild.Shared/MsBuildRule.cs]: Thanks.

**Review by hvitved — APPROVED** [review]

**Review by michaelnebel — APPROVED** [review]
LGTM!

**Pull request merged**